ducc/webhook/registry-listener.go

```diff
@@ -75,70 +75,97 @@ func ExecDucc(msg string, logfile_name string, repository_name string) {
     ima_split := strings.Split(image, "/")
     dkrepo := ima_split[len(ima_split)-1]
 
-    if action == "push" {
+    nOfE := 0
+    repeat := true
 
-        nOfE := 0
-        repeat := true
-        lf_name := ""
+    for repeat {
+       nOfE++
+       repeat = false
+       currentTime := time.Now()
+       timestamp := currentTime.Format("060102-150405")
+       lf_name := logfile_name + "_" + dkrepo + "_" + timestamp
 
-        for repeat {
-          nOfE++
-          repeat = false
-          currentTime := time.Now()
-          timestamp := currentTime.Format("060102-150405")
-          lf_name = logfile_name + "_" + dkrepo + "_" + timestamp
+       if action == "push" {
           fmt.Printf("[DUCC conversion n.%d for %s started...]\n", nOfE, image)
-
           _, err := exec.Command("sudo", "cvmfs_ducc", "convert-single-image", "-n", lf_name, "-p", image, repository_name, "--skip-thin-image", "--skip-podman").Output()
           if err != nil {
             log.Fatal(err)
           }
-
-          // Open the JSON file for reading
-          _, chmodErr := exec.Command("sudo", "chmod", "0755", lf_name).Output()
-          if chmodErr != nil {
-                fmt.Println("Error executing chmod:", chmodErr)
-                return
-          }
-
-          file, fileErr := os.Open(lf_name)
+       } else if action == "delete" {
+          fmt.Printf("[DUCC garbage collection n.%d for %s started...]\n", nOfE, image)
+          _, err := exec.Command("sudo", "cvmfs_server", "transaction", repository_name).Output()
           if err != nil {
-                fmt.Println("Error opening file:", fileErr)
-                return
+            log.Fatal(err)
           }
-          defer file.Close()
-
-          // Create a scanner to read the file line by line
-          scanner := bufio.NewScanner(file)
-
-          // Loop through each line in the file
-          for scanner.Scan() {
-
-                line := scanner.Text()
-                // Parse the line as a JSON object
-                var data map[string]interface{}
-                if err := json.Unmarshal([]byte(line), &data); err != nil {
-                        fmt.Printf("Error parsing JSON: %v\n", err)
-                        continue
-                }
-
-                // Check if "status" is "error"
-                status, exists := data["status"]
-                if exists && status == "error" {
-                        // Perform some action when "status" is "error"
-                        fmt.Printf("[DUCC conversion n.%d failed for layer %s]\n", nOfE, data["layer"])
-                        repeat = true
-                        break;
-                }
+          image_path := "/cvmfs/" + repository_name + "/" + image
+          image_manifest := "/cvmfs/" + repository_name + "/.metadata/" + image
+          _, rmErr := exec.Command("sudo", "rm", "-rf", image_path, image_manifest).Output()
```

Comment:
Here we should either add an assert that `image_path` and `image_manifest` are really absolute paths to cvmfs - the consequences of a future bug inadvertently changing those strings in the future could be bad :)

I think it should be possible and easier to use `cvmfs_server ingest -d` instead.

Reply:
i do agree, but when i tried the ingest -d with this file:

```
ls -l /cvmfs/unpacked.infn.it/harbor.cloud.infn.it/unpacked/almalinux:latest
lrwxrwxrwx. 1 root root 79 Sep 16 13:02 /cvmfs/unpacked.infn.it/harbor.cloud.infn.it/unpacked/almalinux:latest -> ../../.flat/fe/fe9652e6c62e5ff85af09ae7cd2cea7d9d8389baf6977ce90a12867c28ad21be

```
i got:
```
$ sudo cvmfs_server ingest -d harbor.cloud.infn.it/unpacked/almalinux:latest unpacked.infn.it
Using auto tag 'generic-2024-09-25T13:57:30Z'
Processing changes...
[WARNING] 'harbor.cloud.infn.it/unpacked/almalinux' cannot be deleted. Unrecognized file type.
[WARNING] 'latest' cannot be deleted. Unrecognized file type.
Waiting for upload of files before committing...
Committing file catalogs...
Wait for all uploads to finish
Exporting repository manifest
Statistics stored at: /var/spool/cvmfs/unpacked.infn.it/stats.db
Tagging unpacked.infn.it
Flushing file system buffers
Signing new manifest
Remounting newly created repository revision
```
and the file is still there:
```
$ ls -l /cvmfs/unpacked.infn.it/harbor.cloud.infn.it/unpacked/almalinux:latest
lrwxrwxrwx. 1 root root 79 Sep 16 13:02 /cvmfs/unpacked.infn.it/harbor.cloud.infn.it/unpacked/almalinux:latest -> ../../.flat/fe/fe9652e6c62e5ff85af09ae7cd2cea7d9d8389baf6977ce90a12867c28ad21be
```